File: src/test/java/com/ddimensia/java/LibraryImplTest.java
package com.ddimensia.java;

import com.ddimensia.java.example.ExposedImpl;
import com.ddimensia.java.example.LibraryImpl;
import org.junit.Test;

/**
 *
 */
public class LibraryImplTest {
    @Test
    public void testCallHidden() {
        LibraryImpl library = new LibraryImpl();
        ExposedImpl exposed = new ExposedImpl();
        library.acceptHidden(exposed);
    }

    @Test
    public void testCallExposed() {
        LibraryImpl library = new LibraryImpl();
        ExposedImpl exposed = new ExposedImpl();
        library.acceptsImpl(exposed);
    }
}

File: src/test/kotlin/com/ddimensia/kotlin/LibraryImplTest.kt
package com.ddimensia.kotlin

import com.ddimensia.java.example.ExposedImpl
import com.ddimensia.java.example.LibraryImpl
import org.junit.Test

/**
 *
 */
class LibraryImplTest {
    @Test
    fun testCallHidden() {
        val library = LibraryImpl()
        val exposed = ExposedImpl()
        library.acceptHidden(exposed)
    }

    @Test
    fun testCallExposed() {
        val library = LibraryImpl()
        val exposed = ExposedImpl()
        library.acceptsImpl(exposed)
    }
}
